src/lib/SapphireClient.ts

```diff
@@ -107,8 +121,27 @@ export class SapphireClient extends Client {
 	 * @retrun Token of the account used.
 	 */
 	public async login(token?: string) {
+		for (const plugin of SapphireClient.plugins.values(PluginHook.PreLogin)) {
+			plugin.hook.call(this);
+			this.emit(Events.PluginLoaded, plugin.type, plugin.name);
+		}
+
 		await Promise.all([...this.stores].map((store) => store.loadAll()));
-		return super.login(token);
+		const login = await super.login(token);
+
+		for (const plugin of SapphireClient.plugins.values(PluginHook.PostLogin)) {
+			plugin.hook.call(this);
+			this.emit(Events.PluginLoaded, plugin.type, plugin.name);
+		}
+
+		return login;
+	}
+
+	public static plugins = new PluginManager();
+
+	public static use(plugin: typeof Plugin) {
+		this.plugins.use(plugin);
+		return SapphireClient;
```

Comment:
Shouldn't this return.. `this`

Reply:
`SapphireClient` and `this` are the same in this context. Both return the class.

Reply:
If I extend the client, use won't return my client but SapphireClient

Reply:
Understandable.